business/istio_status.go

```diff
@@ -61,8 +62,13 @@ func (iss *IstioStatusService) GetStatus(ctx context.Context) (kubernetes.IstioC
 		return kubernetes.IstioComponentStatus{}, nil
 	}
 
-	if istioStatus, ok := iss.cache.GetIstioStatus(); ok {
-		return istioStatus, nil
+	// In autodiscovery mode, gateway health status can change independently and gateways
+	// use a separate cache with different expiration. To avoid showing stale gateway health,
+	// we always recompute status in autodiscovery mode to get fresh gateway data.
+	if len(iss.conf.ExternalServices.Istio.ComponentStatuses.Components) > 0 {
+		if istioStatus, ok := iss.cache.GetIstioStatus(); ok {
+			return istioStatus, nil
+		}
 	}
```

Comment:
I'm not sure about this. If I understand correctly, "autodiscovery mode" is when no explicit Istio components are configured.  And in that situation you want to always bypass the istioStatus cache because it may not be up to date for all components (gateways in particular). Bypassing the cache and re-computing each time seems like a potential perf issue?

Why do we have Gateway status in different places? Can we consolidate into one cache, or at least keep the caches in sync?

Reply:
Fixed the Mesh and Overview pages to send '/status' requests. The issue was that these pages did not contain Namespaces selector, to which was tightened reloading of status, now it reloads status while entering those pages.